larpix/io/__init__.py

```diff
@@ -1,21 +1,46 @@
+from larpix import configs
+
 class IO(object):
     '''
     Base class for IO objects that explicitly describes the necessary functions
     required by any IO class implementation. Additional functions are not used
     by the larpix core classes.
 
     '''
+    _valid_config_types = ['io']
+    _valid_config_classes = ['IO']
+
     def __init__(self):
         '''
         Declaration of IO object
 
-        :ivar is_listening: flag for ``start_listening`` and ``stop_listening`` commands
+        :ivar is_listening: flag for ``start_listening`` and ``stop_listening``
+        :ivar default_filepath: default configuration path to load
 
         '''
+        self.default_filepath = 'io/default.json'
+
         self.is_listening = False
+        self._io_group_table = {}
```

Comment:
Use [bidict](https://bidict.readthedocs.io/en/latest/home.html) to implement the lookup and just call it ``self._io_group_table``.